integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -1436,4 +1439,206 @@ private String getHostNameForCurl() throws Exception {
+
+    StringBuffer cmdKubectlSh = new StringBuffer("kubectl -n ");
+    cmdKubectlSh
+        .append(domainNS)
+        .append(" exec -it ")
+        .append(domainUid)
+        .append("-")
+        .append(adminServerName)
+        .append(" -- bash -c 'chmod +x -R ")
+        .append(appLocationInPod)
+        .append(" && sh ")
+        .append(appLocationInPod)
+        .append("/")
+        .append(scriptName)
+        .append(" ")
+        .append(nodeHost)
+        .append(" ")
+        .append(nodePort)
+        .append(" ")
+        .append(username)
+        .append(" ")
+        .append(password)
+        .append(" ")
+        .append(appLocationInPod)
+        .append("/")
+        .append(webappName)
+        .append(" ")
+        .append(webappName)
+        .append(" ")
+        .append(clusterName)
+        .append(" ")
+        .append(infoDirNames)
+        .append(" ")
+        .append(archiveExt)
+        .append("'");
+
+    logger.info("Command to exec script file: " + cmdKubectlSh);
+    ExecResult result = ExecCommand.exec(cmdKubectlSh.toString());
+
+    String resultStr =
+        "Command= '"
+            + cmdKubectlSh
+            + "'"
+            + ", exitValue="
+            + result.exitValue()
+            + ", stdout='"
+            + result.stdout()
+            + "'"
+            + ", stderr='"
+            + result.stderr()
+            + "'";
+
+    if (!resultStr.contains("Unable to use a TTY") && result.exitValue() != 0) {
+      throw new RuntimeException("FAILURE: webapp deploy failed - " + resultStr);
+    }
+  }
+
+  /**
+   * Create dir to save Web App files Copy the shell script file and all App files over to the admin
+   * pod Run the shell script to build .war file and deploy the App in the admin pod
+   *
+   * @param appName - Java App name to be deployed
+   * @param scriptName - a shell script to build .war file and deploy the App in the admin pod
+   * @param username - weblogic user name
+   * @param password - weblogc password
+   * @param args - by default, a WAR file is created for a Web App and a EAR file is created for EJB
+   *     App. this varargs gives a client a chance to change EJB's archive extenyion to JAR
+   * @throws Exception
+   */
+  public void buildDeployJavaAppInPod(
+      String appName, String scriptName, String username, String password, String... args)
+      throws Exception {
+    String adminServerPod = domainUid + "-" + adminServerName;
+
+    String appLocationOnHost = BaseTest.getAppLocationOnHost() + "/" + appName;
+    String appLocationInPod = BaseTest.getAppLocationInPod() + "/" + appName;
+    String scriptPathOnHost = BaseTest.getAppLocationOnHost() + "/" + scriptName;
+    String scriptPathInPod = BaseTest.getAppLocationInPod() + "/" + scriptName;
+
+    final String initInfoDirName = "WEB-INF";
+    String archiveExt = "war";
+    String infoDirName = initInfoDirName;
+    File appFiles = new File(appLocationOnHost);
+
```

Comment:
provide code comments, it will be easy for others to understand what this code segment is for

Reply:
on the way...

Reply:
done